src/switch_ivr_originate.c

```diff
@@ -1931,6 +1931,43 @@ static void *SWITCH_THREAD_FUNC early_thread_run(switch_thread_t *thread, void *
 				}
 
 				if (!state->ringback->asis) {
+					switch_codec_implementation_t peer_read_impl = { 0 }, write_impl = { 0 };
+
+					if (switch_core_session_get_read_impl(originate_status[array_pos].peer_session, &peer_read_impl) == SWITCH_STATUS_SUCCESS 
+						&& switch_core_session_get_write_impl(state->oglobals->session, &write_impl) == SWITCH_STATUS_SUCCESS) {
+
+						// sync sample rate mismatch for bridge_early_media
+						if ((peer_read_impl.actual_samples_per_second != write_impl.actual_samples_per_second && peer_read_impl.actual_samples_per_second != state->write_frame->codec->implementation->actual_samples_per_second)
+								|| (state->write_frame->codec->implementation->actual_samples_per_second > write_impl.actual_samples_per_second)) {
+							
+							switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(state->oglobals->session), SWITCH_LOG_DEBUG,
+												"Changing sampling rate from %uhz to %uhz\n", write_impl.actual_samples_per_second, peer_read_impl.actual_samples_per_second);
+							
+							if (switch_core_codec_ready(state->write_codec)) {
```

Comment:
I don't see this write_codec in `early_state_t` struct. Probably a missing commit?

Reply:
Yeah, added. Thanks. Downsides of moving code from one system to another